libs/datamodel/core/src/transform/ast_to_dml/validate.rs

```diff
@@ -177,10 +172,10 @@ impl<'a> Validator<'a> {
         }
     }
 
-    fn validate_names(&self, ast_schema: &ast::SchemaAst, names: &Names<'_>) -> Result<(), Diagnostics> {
+    fn validate_names(&self, db: &ParserDatabase<'_>) -> Result<(), Diagnostics> {
```

Comment:
Same here. Maybe if the struct holds a mutable reference to the database. But then again for me this whole struct feels weird, why do we need an object? Wouldn't normal functions just be enough (and easier to reason about)?

Reply:
I agree. I think this file will be touched when we add relations to `ParserDatabase`.